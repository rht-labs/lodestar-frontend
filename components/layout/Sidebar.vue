<template>
  <div class="pf-c-page__sidebar pf-m-dark">
    <div class="pf-c-page__sidebar-body">
      <Navigation />
    </div>
  </div>
</template>

<script>
import Navigation from '~/components/layout/Navigation.vue'

export default {
  components: {
    Navigation
  }
}
</script>
