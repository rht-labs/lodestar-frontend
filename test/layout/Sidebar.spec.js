import { shallowMount, createLocalVue } from '@vue/test-utils'
import Sidebar from '@/components/layout/Sidebar.vue'

const localVue = createLocalVue()

localVue.component('nuxt-link', {
  props: ['to'],
  template: '<a href="#"><slot>NuxtLink</slot></a>'
})

describe('Sidebar', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(Sidebar, {
      stubs: ['nuxt-link'],
      localVue
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = shallowMount(Sidebar, {
      stubs: ['nuxt-link'],
      localVue
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
