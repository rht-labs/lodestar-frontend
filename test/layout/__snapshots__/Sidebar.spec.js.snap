// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders properly 1`] = `"<div class=\\"pf-c-page__sidebar pf-m-dark\\"><div class=\\"pf-c-page__sidebar-body\\"><navigation-stub></navigation-stub></div></div>"`;
